### PR TITLE
Stack class with tests

### PR DESCRIPTION
**Pull request by sdweber422**

### CODE REVIEW

File: spec/stack.js [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
 import chaiChange from 'chai-change'
 import Stack from '../src/stack'
 
 chai.use(chaiChange)
 
 describe('Stack', () => {
   'use strict'
+  let myStack
+  beforeEach( () => {
+    myStack = new Stack()
```
Comment (EthanJStark) [inline spec/stack.js:11]: Good idea--haven't seen anyone use this on this project before.

File: spec/stack.js [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
     expect(Stack).to.be.a('function')
   })
 
-  context('push()', () => {
+  context('#push()', () => {
     it('pushes an element to the top of the stack.', () => {
-      const myStack = new Stack()
-
       expect(() => myStack.push('foo'))
         .to.alter(() => myStack.length(), { from: 0, to: 1 })
```
Comment (EthanJStark) [inline spec/stack.js:21]: Consider testing additional properties (if not for Stack, then on future data structures). For instance, push might test:

```js
it('changes the top element to the value pushed.', () => {
      const myStack = new Stack()

      expect(() => myStack.push('bar'))
        .to.alter(() => myStack.peek(), { from: null, to: 'bar' })
    })
```

I was proud of the (tests we wrote)[https://github.com/EthanJStark/core-data-structures], if you ever wanna take a peek.

File: spec/stack.js [hunk 3]
```diff
@@ -1,22 +1,53 @@
...
     })
   })
+
+  context('#pop()', () => {
```
Comment (EthanJStark) [inline spec/stack.js]: You might test that the element is no longer contained in myStack after a pop()

Comment (EthanJStark) [reply spec/stack.js]: Also, test a case when one pops on an empty Stack.

**Review by EthanJStark — APPROVED** [review]
Off to a great start, Steve! See comments inline.

File: spec/stack.js (hunk 3 above)
Comment (sdweber422) [reply spec/stack.js]: @EthanJStark Great feedback. I will incorporate this now.
